### PR TITLE
[flutter_tools] opt all flutter tool libraries and tests out of null safety.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Opt the flutter tool out of null safety.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/flutter_project_metadata_test.dart [hunk 1]
```diff
@@ -83,4 +85,4 @@ project_type: {}
 
     expect(logger.traceText, contains('.metadata project_type version is malformed.'));
   });
-}
\ No newline at end of file
+}
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/flutter_project_metadata_test.dart:88]: Huh.

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/bin/tool_backend.dart [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// @dart = 2.8
+
 // @dart = 2.9
```
Comment (christopherfujino) [inline packages/flutter_tools/bin/tool_backend.dart:7]: Delete?

**Review by christopherfujino — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/bin/tool_backend.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/bin/tool_backend.dart:7]: fixed

**Review by zanderso — APPROVED** [review]

**Pull request merged**